r2/r2/lib/pages/pages.py

```diff
@@ -2875,7 +2875,11 @@ def rendered_comment(comments, gilded=False):
 class SubredditStylesheetSource(Templated):
     """A view of the unminified source of a subreddit's stylesheet."""
     def __init__(self, stylesheet_contents):
-        Templated.__init__(self, stylesheet_contents=stylesheet_contents)
+        raw_images = ImagesByWikiPage.get_images(c.site, "config/stylesheet")
+        images = {name: make_url_protocol_relative(url)
+                  for name, url in raw_images.iteritems()}
+        Templated.__init__(self, images=images,
+            stylesheet_contents=stylesheet_contents)
```

Comment:
I think we ought not to duplicate this.  It might be better to move this into a base class that `SubredditStylesheet` and `SubredditStylesheetSource` both inherit from, e.g. something like

```python
class SubredditStylesheetBase(Templated):
    def __init__(self, stylesheet_contents, **kwargs):
        raw_images = ImagesByWikiPage.get_images(c.site, "config/stylesheet")
        images = {name: make_url_protocol_relative(url)
                  for name, url in raw_images.iteritems()}
        super(SubredditStylesheetBase, self).__init__(
            images=images,
            stylesheet_contents=stylesheet_contents,
            **kwargs
        )

class SubredditStylesheet(SubredditStylesheetBase):
    def __init__(self, site=None, stylesheet_contents=""):
        allow_image_upload = site and not site.quarantine
        super(SubredditStylesheet, self).__init__(
            stylesheet_contents=stylesheet_contents,
            site=site,
            allow_image_upload=allow_image_upload,
        )

class SubredditStylesheetSource(subredditStylesheetBase):
    pass
```